server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java

```diff
@@ -343,6 +343,21 @@ private void parseContainer(XMLExtendedStreamReader reader, PathAddress subsyste
                 }
             }
         }
+
+        addSyntheticHealtheOperation(operations, containerAddress);
+    }
+
+    /**
+     * A {@link HealthResource} is not populated to the XML. However the easiest way to propagate a Health node
+     * (we need to attach Health node to the Cache Container node) and attach metrics to it is to
+     * simply simulate reading it.
+     *
+     * Since we are not persisting it - there is no need to check the namespaces.
+     */
+    private void addSyntheticHealtheOperation(Map<PathAddress, ModelNode> operations, PathAddress containerAddress) {
+        PathAddress healthAddress = containerAddress.append(ModelKeys.HEALTH, ModelKeys.HEALTH_NAME);
```

Comment:
Yes, I also thought this is a bit smelly but couldn't find better way to for WF to register HealthResource (technically call `HealthResource#registerAttributes`) (see https://github.com/infinispan/infinispan/pull/4499/files/fb37662ddd39d4e8f1845999e1257203ca3b7515#diff-3aa0cebf53a46eb936bd575a83d9310bR53). Do you have any other idea that I could try?